text_dedup/utils/ftfy_utils.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# @Date    : 2022-12-26 15:37:44
+# @Author  : Chris Ha (hac541309@gmail.com) Chenghao Mou (mouchenghao@gmail.com)
+
+from typing import Dict
+from typing import List
+
+from ftfy import TextFixerConfig
+from ftfy import fix_and_explain
+from ftfy import fix_encoding
+from ftfy import fix_text
+
+
+def fix_text_batch(batch: Dict[str, List], column: str = "text") -> Dict[str, List]:
```

Comment:
Here is one concern I have:

In this function, the text column is overwritten, which is fine if you want to preprocess your data beforehand, but again that's outside the scope of this project.

But if this is intended for preparing for deduplication, then you don't really need the batch function(s). You can simply add the fixing in the fingerprinting function/stage to maximize CPU usage.

Reply:
It could be either or both.
I did intend for both but I am still not sure the best way to implement it in each case (it's also different for exact and near dedup)
As you can see, these are currently added as helpers to lay the groundwork ATM.

Reply:
Hmmm so if I understand you correctly it is not useful for fingerprinting and for this project isn't dealing preprocessing that the batches aren't necessary in any case?
If that is so I'm happy to remove the batching functions.

Reply:
I am fine with `ftfy` being introduced. We can introduce the fixing on the fly during the fingerprinting. For example:

```
tokens: Set[bytes] = {
        bytes(" ".join(t).lower(), "utf-8") for t in ngrams(NON_ALPHA.split(fix_text(content)), ngram_size, min_length)
    }
```

This keeps all the CPU computation together in one go and leaves no storage/cache footprint, so we don't have to iterate the dataset or create a new text column beforehand. Another point is that this repo shouldn't change the original data but only remove deduplicates.

However, this also leads to the question of how the fixing behaviour is controlled, with so many options `ftfy` offers and settings each script already has. It is probably better to offer a config parsing to digest all the settings in the long run.